share/spack/setup-env.csh

```diff
@@ -19,32 +17,59 @@ if ($?_sp_initializing) then
 endif
 setenv _sp_initializing true
 
-if ($?SPACK_ROOT) then
-    set _spack_source_file = $SPACK_ROOT/share/spack/setup-env.csh
-    set _spack_share_dir   = $SPACK_ROOT/share/spack
-
-    # Command aliases point at separate source files
-    alias spack          'set _sp_args = (\!*); source $_spack_share_dir/csh/spack.csh'
-    alias spacktivate    'spack env activate'
-    alias _spack_pathadd 'set _pa_args = (\!*) && source $_spack_share_dir/csh/pathadd.csh'
-
-    # Set variables needed by this script
-    _spack_pathadd PATH "$SPACK_ROOT/bin"
-    eval `spack --print-shell-vars csh`
-
-    # Set up module search paths in the user environment
-    set tcl_roots = `echo $_sp_tcl_roots:q | sed 's/:/ /g'`
-    set compatible_sys_types = `echo $_sp_compatible_sys_types:q | sed 's/:/ /g'`
-    foreach tcl_root ($tcl_roots:q)
-        foreach systype ($compatible_sys_types:q)
-            _spack_pathadd MODULEPATH "$tcl_root/$systype"
-        end
-    end
+# If SPACK_ROOT is not set, we'll try to find it ourselves.
+# csh/tcsh don't have a built-in way to do this, but both keep files
+# they are sourcing open. We use /proc on linux and lsof on macs to
+# find this script's full path in the current process's open files.
+if (! $?SPACK_ROOT) then
+    # figure out a command to list open files
+    if (-d /proc/$$/fd) then
+        set _sp_lsof = "ls -l /proc/$$/fd"
+    else
+        which lsof > /dev/null
+        if ($? == 0) then
+            set _sp_lsof = "lsof -p $$"
+        endif
+    endif
+
+    # filter this script out of list of open files
+    if ( $?_sp_lsof ) then
```

Comment:
Do you want to remove the space after `(` and before `)` to be consistent with your other `if` statements?